stable_baselines3/common/base_class.py

```diff
@@ -730,6 +730,9 @@ def load(
             # See issue https://github.com/DLR-RM/stable-baselines3/issues/597
             if force_reset and data is not None:
                 data["_last_obs"] = None
+            # `n_envs` must be updated. See issue https://github.com/DLR-RM/stable-baselines3/issues/1018
+            if data is not None:
```

Comment:
could you still add a check in `set_env` that raises an error if `num_envs` is different and that ask to use `load()` instead